src/test/java/jewellery/inventory/integration/AuthIntegrationTest.java

```diff
@@ -0,0 +1,127 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthIntegrationTest extends AuthenticatedIntegrationTestBase {
+
+  @Autowired private UserRepository userRepository;
+  @MockBean private PasswordEncoder passwordEncoder;
+
+  private User testUser;
+
+  private final AuthenticationRequestDto authRequest = new AuthenticationRequestDto();
+  private final HttpHeaders headers = new HttpHeaders();
+
+  @Override
+  @BeforeEach
+  void setup() {
+    userRepository.deleteAll();
+    SecurityContextHolder.clearContext();
+    mockPasswordEncoder();
+    createAndSaveTestUser();
+    when(userDetailsService.loadUserByUsername(anyString())).thenReturn(testUser);
+    setupAuthRequestAndHeaders();
+  }
+
+  @Test
+  void generateTokenSuccessfully() {
+    HttpEntity<AuthenticationRequestDto> requestEntity = new HttpEntity<>(authRequest, headers);
+
+    ResponseEntity<AuthenticationResponseDto> response =
+        testRestTemplate.exchange(
+            getBaseAuthUrl() + "/token",
+            HttpMethod.POST,
+            requestEntity,
+            AuthenticationResponseDto.class);
+
+    assertEquals(HttpStatus.CREATED, response.getStatusCode());
+  }
+
+  @Test
+  void testUnauthorizedAccessWithInvalidToken() {
+    String invalidToken = "invalid_token";
+    addTokenToHeaders(invalidToken);
+    HttpEntity<Void> requestEntity = new HttpEntity<>(headers);
+    ResponseEntity<Void> response =
+        testRestTemplate.exchange(getBaseUserUrl(), HttpMethod.GET, requestEntity, Void.class);
+
+    assertEquals(HttpStatus.UNAUTHORIZED, response.getStatusCode());
+  }
+
+  @Test
+  void testAccessToProtectedEndpointWithValidToken() {
+    String mockToken = generateTokenForUser(testUser);
+    addTokenToHeaders(mockToken);
+    HttpEntity<AuthenticationRequestDto> requestEntity = new HttpEntity<>(authRequest, headers);
+
+    ResponseEntity<String> response =
+        testRestTemplate.exchange(getBaseUserUrl(), HttpMethod.GET, requestEntity, String.class);
+    assertEquals(HttpStatus.OK, response.getStatusCode());
+    String responseBody = response.getBody();
+    assertNotNull(responseBody);
+    assertTrue(responseBody.contains(USER_NAME));
+    assertTrue(responseBody.contains(USER_EMAIL));
+  }
+
+  @Test
+  void testInvalidUrlReturnsUnauthorized() {
+    ResponseEntity<Void> response =
+        testRestTemplate.exchange("/invalid-url", HttpMethod.GET, null, Void.class);
+    assertEquals(HttpStatus.UNAUTHORIZED, response.getStatusCode());
```

Comment:
Removed it. The idea came from a colleague of mine who told me what integration tests they write for their project.. And with spring security I do not know how to make it so it returns 404 Not found instead of unauthorized for not existing endpoints... I think the default behavior is to return always unauthorized. I just tested - tried to hit not existing endpoint and provided valid token in the headers - it returned 401 unauthorized.

Reply:
hm, that's interesting. I wonder why...

Reply:
Probably because we override the default behavior and tell it to return 401 instead of 403 🗡️